packages/zostso/__tests__/__system__/cli/issue/cli.zos-tso.issue.cmd.system.test.ts

```diff
@@ -104,5 +104,31 @@ describe("zos-tso issue command", () => {
             expect(response.status).toBe(0);
             expect(new RegExp(regex, "g").test(response.stdout.toString())).toBe(true);
         });
+
+        it("should successfully issue command = \"time\" with an account with an # in it", async () => {
+            const regex = fs.readFileSync(__dirname + "/__regex__/address_space_response.regex").toString();
+
+            const ZOWE_OPT_BASE_PATH = "ZOWE_OPT_BASE_PATH";
+
+            // if API Mediation layer is being used (basePath has a value) then
+            // set an ENVIRONMENT variable to be used by zowe.
+            if (SYSTEM_PROPS.zosmf.basePath != null) {
+                TEST_ENVIRONMENT_NO_PROF.env[ZOWE_OPT_BASE_PATH] = SYSTEM_PROPS.zosmf.basePath;
+            }
+
+            const response = runCliScript(__dirname + "/__scripts__/as/issue_command_fully_qualified.sh",
+                TEST_ENVIRONMENT_NO_PROF,
+                [
+                    SYSTEM_PROPS.zosmf.host,
+                    SYSTEM_PROPS.zosmf.port,
+                    SYSTEM_PROPS.zosmf.user,
+                    SYSTEM_PROPS.zosmf.pass,
+                    "ACCT#"
```

Comment:
In current form the test fails if used against a system which doesn't allow `ACCT#` as account number.
I was thinking it would be better to read it from SYSTEM_PROPS as well, and implement it in the test definition, something like this:
```
        (isAccountWithHash ? it : it.skip)("should successfully issue command = \"time\" with an account with an # in it", async () => {
```
but I was not able to make is work, because we instantiate the test environment in the `beforeAll()`, and Jest seems to be building all the tests even before this step.
I tried with `isAccountWithHash = systemProps.tso.account.includes("#");` in both `beforeAll()` statements, but doesn't help .... debugger shows that the tests are instantiated before these functions are executed, so `isAccountWithHash` is undefined at this time.
I have also tried to have `systemProps.tso.account.includes("#");` as the direct condition when deciding to use `it` or `it.skip`, but it errors out.

Bottom line is that it would be nice to have it as a conditional test, in case someone actually uses it in the custom_properties.yaml, but we need some creative solution for that.

Reply:
> `(isAccountWithHash ? it : it.skip)("should successfully issue command = \"time\" with an account with an # in it", async () => {`

Really good thinking

Reply:
Update on this.

It works fine if we determine if the account has `#` in it before the parent `describe()` definition. This way, when the tests are initialized, the variable is already set. I did it by reading the `custom_properties.yaml` file directly. Small caveat is that when we have only numbers, it gets cast into `number` type, so the `.includes("#")` function doesn't exit. Had to cast to string to make it work.
The reading of the file was right after the imports, but I don't see why this test can't come from `testUtils` (makes it more elegant).

Biggest showstopper is that we create the test environment dynamically, so this means that this variable might be wrong in case we are not taking the properties from the yaml file.

Still thinking how to make this more foolproof 😄 .

Reply:
Sorry for taking too long to respond here.
After looking at this with a fresh pair of eyes (👀 ), it's seem to be more trouble that the benefit we get out of it (i.e. skipping one test).
However, I don't think it's a bad practice. Maybe we can revisit this in the future to use/update the testing utilities.

Reply:
Silly me. https://github.com/zowe/zowe-cli/pull/658#discussion_r383248657
Sorry for repeating myself.